### PR TITLE
Trust show: parse inline public keys

### PR DESCRIPTION
**Pull request by aweiteka**
Follow-on to #853

### CODE REVIEW

**Comment by aweiteka** [comment]
@rh-atomic-bot retry

**Comment by aweiteka** [comment]
bot, retest this please

File: Atomic/trust.py [hunk 1]
```diff
@@ -478,10 +479,10 @@ def get_gpg_id(self, keys):
         keylist = None
         for key in keys:
             if not os.path.exists(key):
-                # extend when we support parsing inline 'keyData' base64 encoded key
-                return ""
-            else:
-                cmd = ["gpg2", "--with-colons", key]
+                with tempfile.NamedTemporaryFile(delete=False) as tmpkey:
```
Comment (rhatdan) [inline Atomic/trust.py]: I think you need to make sure that this file is created on /run, not somewhere a user might be able to attack

File: Atomic/trust.py [hunk 2]
```diff
@@ -478,12 +479,13 @@ def get_gpg_id(self, keys):
...
-                return ""
-            else:
-                cmd = ["gpg2", "--with-colons", key]
+                with tempfile.TemporaryFile() as tmpkey:
+                    tmpkey.write(b64decode(key))
+                key = tmpkey.name
+            cmd = ["gpg2", "--with-colons", key]
             try:
                 results = util.check_output(cmd).decode('utf-8')
+                tmpkey.close()
```
Comment (aweiteka) [inline Atomic/trust.py]: @rhatdan I tightened the tempfile up a bit. I'm not sure what you mean by using /run. Prefix dir for tempfile?

Comment (rhatdan) [inline Atomic/trust.py]: yes

File: Atomic/trust.py [hunk 3]
```diff
@@ -478,10 +479,10 @@ def get_gpg_id(self, keys):
...
         for key in keys:
             if not os.path.exists(key):
-                # extend when we support parsing inline 'keyData' base64 encoded key
-                return ""
-            else:
-                cmd = ["gpg2", "--with-colons", key]
+                with tempfile.TemporaryFile() as tmpkey:
+                    tmpkey.write(b64decode(key))
+                key = tmpkey.name
+            cmd = ["gpg2", "--with-colons", key]
```
Comment (rhatdan) [inline Atomic/trust.py]: Should we be removing the tmpfile when this completes?

File: Atomic/trust.py [hunk 4]
```diff
@@ -486,17 +487,20 @@ def get_gpg_id(self, keys):
...
             return ""
-        keylist = None
+        (keylist, tmpkey) = None, None
         for key in keys:
             if not os.path.exists(key):
-                # extend when we support parsing inline 'keyData' base64 encoded key
-                return ""
-            else:
-                cmd = ["gpg2", "--with-colons", key]
+                with tempfile.NamedTemporaryFile(delete=False, dir="/run/") as tmpkey:
```
Comment (rhatdan) [inline Atomic/trust.py]: Add delete=True and then when the file is closed it should get deleted.  Then you should not need to do code below.  I think?

Comment (aweiteka) [inline Atomic/trust.py]: `delete=True` (the default) deletes the file when the with-open loop exists, so it's gone when we need it failing the gpg2 cmd.

**Comment by rhatdan** [comment]
Can you merge the pull requests, and give me a test?

**Comment by rhatdan** [comment]
@rh-atomic-bot r+ 0edc5f7

**Pull request closed without merging**